src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -199,17 +190,89 @@ private boolean canGoPlanting(@NotNull Field currentField, @NotNull BuildingFarm
 
         if (shouldTryToGetSeed)
         {
+            int slot = worker.findFirstSlotInInventoryWith(currentField.getSeed());
+            if (slot != -1)
+            {
+                requestSeeds = false;
+            }
             if (walkToBuilding())
             {
                 return false;
             }
-            isInHut(new ItemStack(currentField.getSeed()));
-            shouldTryToGetSeed = false;
+            if (isInHut(new ItemStack(currentField.getSeed())))
+            {
+                requestSeeds = false;
+            }
+            shouldTryToGetSeed = requestSeeds;
+
+            if (shouldTryToGetSeed)
+            {
+                return false;
+            }
         }
 
         return true;
     }
 
+    private AIState cycle()
+    {
+        @Nullable BuildingFarmer buildingFarmer = getOwnBuilding();
+       
+        if (buildingFarmer == null || buildingFarmer.getCurrentField() == null  || checkForHoe())
+        {
+            return AIState.PREPARING;
+        }
+        @Nullable Field field = buildingFarmer.getCurrentField();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'field' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)